uiowa/uiowa
# Feature: Initial uids_base integration with branding site

This is a `WIP` for mapping some components from UIDS to the SiteNow paragraphs.   

# Brand/SiteNow Integration

## How to test
* Testing should be done at https://brand.dev.drupal.uiowa.edu
* You'll need to log in via the Hawkid link at the bottom of the page.

**Accordion**
- Edit any page and create an Accordion paragraph.  Save the page and test that it is working. 

**Article content type**
- Add multiple authors to an article.  Save the page and verify that they are listed on one line with "|" between them.  Add an Organization and Link and verify that it is displaying inline with authors. 
- Add a link to the "source link" field and check "link directly to source" and go to https://brand.dev.drupal.uiowa.edu/news and verify that the list item links directly to that source link.
- Add a featured image to an article.   Verify that it appears below the page title. 
- Change Article display view options on https://brand.dev.drupal.uiowa.edu/ to `default` `grid` and verify that the display changes for each setting.   The `masonry` setting still needs some adjustments. 
- Change Article List settings on https://brand.dev.drupal.uiowa.edu/ to "Hide Summaries" and verify that summary is hidden.  Note: the "Hide Images" options is currently broken: https://github.com/uiowa/uiowa/issues/1414.
- Go to https://brand.dev.drupal.uiowa.edu/news and verify that articles are displayed in a list.   Add multiple authors to an article and verify that it appears in this list.  Note: The hiding of authors in list option is currently not working: https://github.com/uiowa/uiowa/issues/1413
- Note: the mobile breakpoint for the articles list needs to be adjusted https://github.com/uiowa/uiowa/issues/1410

**Cards** 
- Go to https://brand.dev.drupal.uiowa.edu/logo and verify that cards are all appearing equal height if section vertical alignment is set to "none"
- Go to https://brand.dev.drupal.uiowa.edu/logos/secondary-logos and verify that card Image, title, and subtitle appear in the top cards. 
- Note: The card link is broken at this time: https://github.com/uiowa/uiowa/issues/1408
- Note: inline card image alignment padding needs to be adjusted: https://github.com/uiowa/uiowa/issues/1411

**Image Gallery**
- Go to https://brand.dev.drupal.uiowa.edu/imagery and verify that grid image galleries are working. 

**Image alignment**
- Go to https://brand.dev.drupal.uiowa.edu/secondary-logos and verify that Seal is centered
- Go to https://brand.dev.drupal.uiowa.edu/ and verify that small icons within cards are aligned left of the content

**Embedded Video**
- Go to https://brand.dev.drupal.uiowa.edu/ and verify that YouTube video appears

**Container Queries**
- I added https://philipwalton.com/articles/responsive-components-a-solution-to-the-container-queries-problem/ to uids_base.js https://github.com/uiowa/uiowa/blob/f376f68cd77d74510409713d34fc571b60c2c0cf/docroot/themes/custom/uids_base/assets/js/uids_base.js#L8 in order to address the line-height issue when there are 4 or 5-columns of cards in a row like on https://brand.dev.drupal.uiowa.edu/design/color.  If you resize the page down you can see that the line-height of the black and gold cards adjusts.  It is not working for the 4-columns of cards and this feature needs to be adjusted in future issues.  

# Template refactoring

## How to test
* Visit https://uiowa.dev.drupal.uiowa.edu/student-life and scroll down to the "Like no other college town" banner.
* Verify that the look/feel and behavior are the same as on https://uiowa.stage.drupal.uiowa.edu/student-life
